python/cudf/cudf/_lib/pylibcudf/copying.pyx

```diff
@@ -55,3 +66,198 @@ cpdef Table gather(
             )
         )
     return Table.from_libcudf(move(c_result))
+
+
+cpdef Column shift(Column input, size_type offset, Scalar fill_values):
+    cdef unique_ptr[column] c_result
+    with nogil:
+        c_result = move(
+            cpp_copying.shift(
+                input.view(),
+                offset,
+                dereference(fill_values.c_obj)
+            )
+        )
+    return Column.from_libcudf(move(c_result))
+
+
+cpdef Table scatter(object source, Column scatter_map, Table target_table):
+    cdef unique_ptr[table] c_result
+    cdef vector[reference_wrapper[constscalar]] source_scalars
+    cdef Scalar slr
+
+    if isinstance(source, Table):
+        with nogil:
+            c_result = move(
+                cpp_copying.scatter(
+                    (<Table> source).view(),
+                    scatter_map.view(),
+                    target_table.view(),
+                )
+            )
+    elif isinstance(source, list):  # TODO: is list too restrictive?
+        for slr in source:
+            source_scalars.push_back(
+                reference_wrapper[constscalar](dereference(slr.c_obj))
+            )
+
+        with nogil:
+            c_result = move(
+                cpp_copying.scatter(
+                    source_scalars,
+                    scatter_map.view(),
+                    target_table.view(),
+                )
+            )
+    else:
+        raise ValueError("source must be a Table or list[Scalar]")
+
+    return Table.from_libcudf(move(c_result))
+
+
+cpdef object empty_like(object input):
```

Comment:
Is it worth having an `empty_column_like()` and `empty_table_like()`?

Reply:
I debated this as well. I could go either way as long as whatever we decide here is made consistent policy across pylibcudf. The current approach matches the libcudf API and makes it simpler for the user at the cost of having to deal with typing internally and introducing a bit of overhead as a result. The alternative would be strongly typed but slightly more verbose functions. Do you think the latter is better?

Reply:
I think:

- Separate APIs for `Column`/`Table`
- No need for separate APIs for other kinds of overloads in C++

Reply:
I'll also include scalars alongside column and tables, but otherwise yes agreed.

Reply:
Hmm this choice ended up being a bit more loaded than I would have liked given the functions that accept either a table or a vector of scalars. Let me know what you think of the current set of functions that I've defined.

Reply:
Hmm, I agree I don't love it. If this were pure Python maybe `overload`s would be the right choice typing wise.

Reply:
We discussed naming offline. The state of the PR as of https://github.com/rapidsai/cudf/pull/14508/commits/b0ae7f0ff7ca31c35b84e0e87723450f16bc1b69 is our final choice of names.